src/index.js

```diff
@@ -123,18 +123,14 @@ class UglifyJsPlugin {
               let input;
               let inputSourceMap;
 
-              if (this.options.sourceMap) {
-                if (asset.sourceAndMap) {
-                  const sourceAndMap = asset.sourceAndMap();
-                  inputSourceMap = sourceAndMap.map;
-                  input = sourceAndMap.source;
-                } else {
-                  inputSourceMap = asset.map();
-                  input = asset.source();
-                }
+              if (this.options.sourceMap && asset.sourceAndMap) {
+                const sourceAndMap = asset.sourceAndMap();
+                input = sourceAndMap.source;
+                inputSourceMap = sourceAndMap.map;
                 sourceMap = new SourceMapConsumer(inputSourceMap);
               } else {
                 input = asset.source();
+                inputSourceMap = null;
```

Comment:
This is the windows related fix ?

Reply:
@michael-ciniawsky no, when `devtool` set to null we don't have `map`

Reply:
@evilebottnawi Sure, but this is nevertheless the fix for the parallel source maps bug :) ? It doesn't seem related to `options.parallel` exclusively then, should we reframe the commit message ?

Reply:
@michael-ciniawsky in this commit i just simplify logic and add tests to make sure that everything works and the problem is not in the plugin